lib/java/src/org/apache/thrift/transport/TByteBuffer.java

```diff
@@ -0,0 +1,88 @@
+package org.apache.thrift.transport;
+
+import java.nio.BufferOverflowException;
+import java.nio.BufferUnderflowException;
+import java.nio.ByteBuffer;
+
+/**
+ * ByteBuffer-backed implementation of TTransport.
+ */
+public class TByteBuffer extends TTransport {
+  private ByteBuffer byteBuffer;
+
+  /**
+   * Creates a new TByteBuffer wrapping a given NIO ByteBuffer.
+   */
+  public TByteBuffer(ByteBuffer byteBuffer) {
+    this.byteBuffer = byteBuffer;
+  }
+
+  /**
+   * Resets the underlying NIO ByteBuffer.
+   */
+  public void reset(ByteBuffer byteBuffer) {
```

Comment:
Fair point -- I'll replace reset(...) with setByteBuffer(...) and nioBuffer() with getByteBuffer respectively.